DTS fixes for SPI Flash on SC5xxx

ozan956 opened the pull request.

pamolloy committed:
arm: dts: sc5xx: drop SPI flash from common dtsi

SPI flash devices are not common across all SC5xx boards. They
already defined in the SoM-specific dtsi files. Keeping the
definition in the common dtsi is therefore incorrect.

Fixes: 7210fc6 ("board: adi: Add support for SC598")
Signed-off-by: Philip Molloy <philip.molloy@analog.com>
(cherry picked from commit e98887b7e2214552a0eb1a78d241d76052479198)
Signed-off-by: Ozan Durgut <ozan.durgut@analog.com>

pamolloy commented:
CC @gmalysa

pamolloy commented on arch/arm/dts/sc5xx.dtsi:
```diff
@@ -154,17 +154,6 @@
-				spi-rx-bus-width = <4>;
-				spi-max-frequency = <10000000>;
-				bootph-pre-ram;
-			};
```
When cherry picking use `git cherry-pick -xs`, which adds the original commit hash.

In the commit message also note that this patch was partially applied in the series, hence why we need to add it here.

pamolloy commented on arch/arm/dts/sc598-som-revD.dtsi:
```diff
@@ -42,17 +42,49 @@
 		};
 
 		spi2flash-cs {
 			gpio-hog;
```
This should be copied from 2025.07 🤞

pamolloy replied:
And add a `Fixes:`

pamolloy commented on arch/arm/dts/sc598-som-ezkit.dts:
```diff
@@ -79,7 +79,7 @@
 			gpio-hog;
 			gpios = <8 GPIO_ACTIVE_LOW>;
-			output-high;
+			output-low;
```
We can drop this assuming there is no conflict between octal SPI flash and SOM SPI flash

pamolloy replied:
SPI2 and OSPI are muxed on the same pins of the SC598

ozan956 committed:
arm: dts: sc598: add missing GPIO hogs for Rev D

Add missing GPIO hogs for UART0 enable, UART0 flow control, and
eMMC control signals.

Fixes: 7210fc6 ("board: adi: Add support for SC598")
Signed-off-by: Ozan Durgut <ozan.durgut@analog.com>

ozan956 committed:
arm: dts: sc598: disable Octal SPI flash

Disable the OSPI on EZKIT so the SPI flash on the SoM can be
used, since they have conflicting signals.

Fixes: 7210fc6 ("board: adi: Add support for SC598")
Signed-off-by: Ozan Durgut <ozan.durgut@analog.com>

pamolloy commented on arch/arm/dts/sc598-som-revD.dtsi:
```diff
@@ -55,6 +55,38 @@
+			bootph-pre-ram;
+		};
+
+		emmc {
```
https://github.com/analogdevicesinc/u-boot/blob/adi-u-boot-2025.10.y/arch/arm/dts/sc598-som-revE.dtsi#L48

pamolloy commented on arch/arm/dts/sc598-som-revD.dtsi:
```diff
@@ -55,6 +55,38 @@
+			bootph-pre-ram;
+		};
+
+		emmc-som-en {
```
https://github.com/analogdevicesinc/u-boot/blob/adi-u-boot-2025.10.y/arch/arm/dts/sc598-som-revE.dtsi#L55
